File: day1/day1.py
#!/usr/bin/env python3

import sys

if __name__ == "__main__":
    if len(sys.argv) < 2 :
        print("Usage: {} filename".format(sys.argv[0]), file=sys.stderr)
        sys.exit(1)

    try:
        with open(sys.argv[1], "rt") as f:
            txt = f.read()
    except:
        print("Cannot open", sys.argv[1], file=sys.stderr)
        sys.exit(1)

    floor = 0
    found = None
    for i, v in enumerate(txt):
        if v == "(":
            floor += 1
        elif v == ")":
            floor -= 1

        if found is None and floor == -1:
            found = i + 1

    print("Part1:", floor)
    print("Part2:", found)

File: day1/day1.c
#include <stdint.h>
#include <stdio.h>

int main(int argc, char *argv[])
{
	if (argc < 2)
	{
		fprintf(stderr, "Usage: %s filename\n", argv[0]);
		return 1;
	}

	FILE *input = fopen(argv[1], "rb");
	if (!input)
	{
		fprintf(stderr, "Cannot open %s\n", argv[1]);
		return 1;
	}


	int c;
	size_t pos = SIZE_MAX;
	int floor = 0;
	for (size_t i = 1; (c = getc(input)) != EOF; i++)
	{
		switch(c)
		{
		case '(': floor++; break;
		case ')': floor--; break;
		default:
		break;
		}

		if (floor == -1 && pos == SIZE_MAX)
		{
			pos = i;
		}
	}
	fclose(input);

	printf("Part1: %d\n", floor);
	printf("Part2: %zu\n", pos);
	return 0;
}
